tests/evaluate/test_evaluation.py

```diff
@@ -2139,3 +2139,25 @@ def test_import_evaluation_dataset():
     # This test is to validate both imports work at the same time
     from mlflow.models.evaluation import EvaluationDataset
     from mlflow.models.evaluation.base import EvaluationDataset  # noqa: F401
+
+
+def test_evaluate_show_error_stack(linear_regressor_model_uri, diabetes_dataset):
+    mlflow.pyfunc.load_model(linear_regressor_model_uri)
+
+    with mlflow.start_run():
+        with mock.patch(
+                "mlflow.pyfunc.scoring_server.client.ScoringServerClient.wait_server_ready"
+        ) as mock_wait_server_ready, mock.patch(
+            "subprocess.Popen.communicate",
```

Comment:
can we patch the `serve` method instead? Patching built-in libs often leads unintended impacts.

Reply:
just saw https://github.com/mlflow/mlflow/pull/12473#issuecomment-2193877338

Reply:
why patching `wait_server_ready` works but patching `serve` doesn't work?

Reply:
`serve` runs in child process spawned by `subprocess.Popen`, but the mock.patch is executed in the parent process.

Reply:
```python
    if is_port_connectable:
        server_port = find_free_port()
        # patching this should work as well, right?
        scoring_server_proc = pyfunc_backend.serve(
            model_uri=local_path,
            port=server_port,
            host="127.0.0.1",
            timeout=MLFLOW_SCORING_SERVER_REQUEST_TIMEOUT.get(),
            enable_mlserver=False,
            synchronous=False,
            stdout=subprocess.PIPE,
            stderr=subprocess.STDOUT,
        )
        client = ScoringServerClient("127.0.0.1", server_port)
    else:
        scoring_server_proc = pyfunc_backend.serve_stdin(local_path)
        client = StdinScoringServerClient(scoring_server_proc)

    _logger.info(f"Scoring server process started at PID: {scoring_server_proc.pid}")
    try:
        # If patching this works,
        client.wait_server_ready(timeout=90, scoring_server_proc=scoring_server_proc)
    except Exception as e:
        raise MlflowException("MLflow model server failed to launch.") from e

```